internal/provider/resource_tfe_organization.go

```diff
@@ -192,9 +199,22 @@ func resourceTFEOrganizationUpdate(d *schema.ResourceData, meta interface{}) err
 		options.CostEstimationEnabled = tfe.Bool(costEstimationEnabled.(bool))
 	}
 
-	// If send_passing_statuses_for_untriggered_speculative_plans is supplied, set it using the options struct.
-	if sendPassingStatusesForUntriggeredSpeculativePlans, ok := d.GetOk("send_passing_statuses_for_untriggered_speculative_plans"); ok {
-		options.SendPassingStatusesForUntriggeredSpeculativePlans = tfe.Bool(sendPassingStatusesForUntriggeredSpeculativePlans.(bool))
+	if d.HasChange("send_passing_statuses_for_untriggered_speculative_plans") {
+		// If send_passing_statuses_for_untriggered_speculative_plans is supplied, set it using the options struct.
+		if sendPassingStatusesForUntriggeredSpeculativePlans, ok := d.GetOk("send_passing_statuses_for_untriggered_speculative_plans"); ok {
+			options.SendPassingStatusesForUntriggeredSpeculativePlans = tfe.Bool(sendPassingStatusesForUntriggeredSpeculativePlans.(bool))
+		} else {
+			options.SendPassingStatusesForUntriggeredSpeculativePlans = tfe.Bool(false)
+		}
+	}
+
+	if d.HasChange("aggregated_commit_status_enabled") {
+		// If aggregated_commit_status_enabled is supplied, set it using the options struct.
+		if aggregatedCommitStatusEnabled, ok := d.GetOk("aggregated_commit_status_enabled"); ok {
+			options.AggregatedCommitStatusEnabled = tfe.Bool(aggregatedCommitStatusEnabled.(bool))
+		} else {
+			options.AggregatedCommitStatusEnabled = tfe.Bool(false)
```

Comment:
Just for my understanding, when will this condition be met, and why do we set it to `false` in that case? Like the attribute was set previously in the terraform config but is removed in the new config?

Reply:
That's what I assumed at first! 

I need to do some additional testing to make sure I just didn't have some weird symlink issues debugging.

What was happening was previous apply with  `aggregated_commit_status_enabled = true` then changing it to `aggregated_commit_status_enabled = false` would not satisfy the d.GetOkay("aggregated_commit_status_enabled") block. Which I don't think is a expected behavior.

Reply:
Addresses this concern: https://github.com/hashicorp/terraform-provider-tfe/pull/1169#discussion_r1469802937